sql-plugin/src/main/scala/org/apache/spark/sql/rapids/mathExpressions.scala

```diff
@@ -419,6 +419,81 @@ case class GpuCot(child: Expression) extends GpuUnaryMathExpression("COT") {
   }
 }
 
+case class GpuHypot(left: Expression, right: Expression) extends CudfBinaryMathExpression("HYPOT") {
+
+  override def binaryOp: BinaryOp = BinaryOp.ADD
+
+  // naive implementation of hypot
+  // r = sqrt(lhs^2 + rhs^2) 
+  // This is prone to overflow with regards either square term
+  // However, we can reduce it to
+  // r = sqrt(x^2 + y^2) = x * sqrt(1 + (y/x)^2)
+  // where x = max(abs(lhs), abs(rhs)), y = min(abs(lhs), abs(rhs))
+  // which will only overflow if both terms are near the maximum representation space
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
+    // Using this form, because we really need a max and min that aren't NULL-aware
```

Comment:
This is large and complicated enough of a method that I think it would be better if we broke it up into smaller functions in an object.  Just to make the code more readable, and simpler to reason about.